engines/python/setup/djl_python/neuron_utils/model_loader.py

```diff
@@ -0,0 +1,338 @@
+        if os.path.isdir(model_path):
+            load_path = model_path
+        else:
+            load_path = self.init_load_path()
+        return load_path
+
+    @abstractmethod
+    def load_model(self):
+        """Model loading method which returns a loaded model"""
+
+    @abstractmethod
+    def partition(self, save_path, tokenizer):
+        """Model compilation and export method which returns a loaded model"""
+
+
+class TNXModelLoader(ModelLoader):
+    """Model loader and compiler for legacy compiled or HuggingFace precompiled artifacts"""
+    MODEL_TYPE_TO_CLS_LOADER = {
+        "opt": "opt.model.OPTForSampling",
+        "gpt2": "gpt2.model.GPT2ForSampling",
+        "gptj": "gptj.model.GPTJForSampling",
+        "gpt_neox": "gptneox.model.GPTNeoXForSampling",
+        "llama": "llama.model.LlamaForSampling",
+        "bloom": "bloom.model.BloomForSampling"
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model = None
+        self.load_path = None
+        self.split_model_path = None
+        self.compiled_graph_path = None
+        module_name, class_name = self.MODEL_TYPE_TO_CLS_LOADER[
+            self.model_config.model_type].rsplit(".", maxsplit=1)
+        module = importlib.import_module(f"transformers_neuronx.{module_name}")
+        self._neuronx_class = getattr(module, class_name, None)
+        if self._neuronx_class is None:
+            raise ImportError(
+                f"{class_name} not found in {module_name}. Please check transformers-neuronx version."
+            )
+
+    @staticmethod
+    def get_neuronxcc_version():
+        global _neuronxcc_version
+        if _neuronxcc_version is not None:
+            return _neuronxcc_version
+        try:
+            import neuronxcc
+        except ImportError:
+            raise ModuleNotFoundError(
+                "NeuronX Compiler python package is not installed.")
+        _neuronxcc_version = neuronxcc.__version__
+        return _neuronxcc_version
+
+    def get_model_specific_kwargs(self):
+        model_kwargs = {
+            "batch_size": self.config.batch_size,
+            "amp": self.config.amp,
+            'tp_degree': self.config.tensor_parallel_degree,
+            "n_positions": self.config.n_positions,
+            "unroll": self.config.unroll
+        }
+        if self.model_config.model_type == "llama":
+            model_kwargs[
+                'context_length_estimate'] = self.config.context_length_estimate
+        return model_kwargs
+
+    def update_model_config_to_neuron(self):
+        neuron_config = {
+            "neuron": {
+                "auto_cast_type": self.config.amp,
+                "batch_size": self.config.batch_size,
+                "compiler_type": "neuronx-cc",
+                "compiler_version": self.get_neuronxcc_version(),
+                "num_cores": self.config.tensor_parallel_degree,
```

Comment:
what is this compiler config for?

Reply:
This is used to match the AutoConfig `config.json` that comes out of an optimum compiled model so that a model compiled using the TNXModelLoader can be loaded using the OptimumModelLoader and vice versa.